src/template/block.rs

```diff
@@ -0,0 +1,121 @@
+use super::span::{ByteSpan, Spanned};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
some documentation would be great here

Reply:
Documentation will be added after merge on a separate branch. As currently, there is no/little documention, this step will be done in one go.